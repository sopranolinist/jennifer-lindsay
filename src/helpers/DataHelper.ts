import get from 'lodash/get';
// import { DataValue } from 'react-apollo';
// import * as API from '../API';

export const NO_SUPERVISOR = 'NO SUPERVISOR';
export const CONTRACTOR_PTO_BALANCE = 0;

export enum testAccounts {
  EXECUTIVE = 'dhq.executive@dws.la',
  OPERATIONS_MANAGER = 'dhq.operations_manager@dws.la',
  PROJECT_MANAGER = 'dhq.project_manager@dws.la',
  EMPLOYEE = 'dhq.employee@dws.la',
  CONTRACTOR = 'dhq.contractor@dws.la',
}

export const isTestAccount = (workEmail: string | undefined) => {
  return (
    workEmail === testAccounts.EXECUTIVE ||
    workEmail === testAccounts.OPERATIONS_MANAGER ||
    workEmail === testAccounts.PROJECT_MANAGER ||
    workEmail === testAccounts.EMPLOYEE ||
    workEmail === testAccounts.CONTRACTOR
  );
};

export enum actionChoices {
  ADD = 'Add',
  EDIT = 'Edit',
  NONE = '',
}

export enum objectChoices {
  PERSON = 'Person',
  POSITION = 'Position',
  TITLE = 'Title',
  NONE = '',
}

export enum formResources {
  POSITION = 'position',
  TITLE = 'title',
  PERSON = 'person',
}

export enum formModes {
  UPDATE = 'update',
  CREATE = 'create',
}

// export type PersonItem = NonNullable<
//   NonNullable<API.ListPersonsQuery['listPersons']>['items']
// >[number];

// export type PositionItem = NonNullable<
//   NonNullable<API.ListPositionsQuery['listPositions']>['items']
// >[number];

// export type TitleItem = NonNullable<
//   NonNullable<API.ListTitlesQuery['listTitles']>['items']
// >[number];

export const formatFormDate = (date: string): string => {
  return date.length > 0 ? `${date.substr(6, 4)}-${date.substr(0, 2)}-${date.substr(3, 2)}` : '';
};

export const dateRangeOverlap = (
  startDate1: Date,
  endDate1: Date,
  startDate2: Date,
  endDate2: Date,
): boolean => {
  return (
    (startDate1 >= startDate2 && startDate1 <= endDate2) ||
    (endDate1 >= startDate2 && endDate1 <= endDate2) ||
    (startDate1 <= startDate2 && endDate1 >= endDate2)
  );
};

export const dateString = (date: Date): string => {
  const today = new Date(`${date.toISOString().substr(0, 10)}T00:00:00.000Z`);
  return today.toISOString();
};

// export const requestIsOpen = (item: API.CreateTimeoffInput | null) => {
//   // Open := PENDING or APPROVED, Closed := CANCELLED or DENIED (or other invalid status value)
//   return item?.currentStatus === 'PENDING' || item?.currentStatus === 'APPROVED';
// };

export const convertBoolToString = (input: boolean | null): string => {
  // eslint-disable-next-line no-nested-ternary
  return input === true ? 'true' : input === false ? 'false' : '';
};

export const convertStringToBool = (input: string): boolean | null => {
  // eslint-disable-next-line no-nested-ternary
  return input === 'true' ? true : input === 'false' ? false : null;
};

// export const calcPersonInitialState = (
//   person: PersonItem,
//   positions: PositionItem[],
//   id: string,
// ): any => {
//   return {
//     firstName: get(person, 'firstName', ''),
//     lastName: get(person, 'lastName', ''),
//     workEmail: get(person, 'workEmail', ''),
//     personalEmail: get(person, 'personalEmail', ''),
//     role: get(person, 'role'),
//     active: convertBoolToString(get(person, 'active', null)),
//     billable: convertBoolToString(get(person, 'billable', null)),
//     position: positions.find((position: PositionItem) => get(position, 'person.id') === id),
//     birthdate: get(person, 'birthdate', ''),
//     address: get(person, 'address', ''),
//     city: get(person, 'city', ''),
//     state: get(person, 'state', ''),
//     zip: get(person, 'zip', ''),
//     phone: get(person, 'phone', ''),
//     emergencyContactName: get(person, 'emergencyContactName', ''),
//     emergencyContactEmail: get(person, 'emergencyContactEmail', ''),
//     emergencyContactPhone: get(person, 'emergencyContactPhone', ''),
//     emergencyContactRelation: get(person, 'emergencyContactRelation', ''),
//     startDate: get(person, 'startDate', ''),
//     endDate: get(person, 'endDate', ''),
//     hourlyRate: get(person, 'hourlyRate', ''),
//     id,
//   };
// };

export const formatTitle = (title: any): string => `${get(title, 'name', 'TBD')}`;

export const formatPerson = (person: any): string =>
  `${get(person, 'lastName')}, ${get(person, 'firstName')}`;

export const formatPosition = (position: any): string => {
  const person = position.person ? formatPerson(position.person) : `TBD`;
  const title = get(position, 'title.name', 'TBD');
  const supervisorPerson = get(position, 'directSupervisor.person')
    ? formatPerson(position.directSupervisor.person)
    : `N/A`;

  const supervisorTitle = get(position, 'directSupervisor.title.name', 'TBD');
  const supervisor = get(position, 'directSupervisor.person')
    ? `${supervisorTitle} (${supervisorPerson})`
    : 'N/A';
  return `${title} (${person}) - Supervisor: ${supervisor}`;
};

// export const getPositions = (
//   dataListPositions: DataValue<API.ListPositionsQuery, API.ListPositionsQueryVariables>,
// ): any[] => {
//   return dataListPositions &&
//     dataListPositions.listPositions &&
//     dataListPositions.listPositions.items
//     ? dataListPositions.listPositions.items.sort((a: any, b: any) => {
//         if (a.title && a.title.name && b.title && b.title.name) {
//           return a.title.name < b.title.name ? -1 : a.title.name > b.title.name ? 1 : 0;
//         }
//         return 0;
//       })
//     : [];
// };

// export const getTitleItems = (
//   dataListTitles: DataValue<API.ListTitlesQuery, API.ListTitlesQueryVariables>,
// ): any[] => {
//   return dataListTitles && dataListTitles.listTitles && dataListTitles.listTitles.items
//     ? dataListTitles.listTitles.items.sort((a: any, b: any) => {
//         return a.name < b.name ? -1 : a.name > b.name ? 1 : 0;
//       })
//     : [];
// };

// export const getTitleNames = (titleItems: TitleItem[]): string[] => {
//   return titleItems
//     ? titleItems
//         .map((item: TitleItem) => {
//           return item && item.name ? item.name : '';
//         })
//         .filter((item: string) => item.length > 0)
//         .sort()
//     : [];
// };

// export const getDirectSupervisors = (
//   dataListPersons: DataValue<API.ListPersonsQuery, {}>,
// ): any[] => {
//   const dsItems =
//     dataListPersons && dataListPersons.listPersons && dataListPersons.listPersons.items
//       ? dataListPersons.listPersons.items
//       : [];

//   return dsItems
//     .map((item: any) => {
//       return { id: item.id, name: `${item.lastName}, ${item.firstName}` };
//     })
//     .sort((a: any, b: any) => {
//       return a.name < b.name ? -1 : a.name > b.name ? 1 : 0;
//     })
//     .concat({ id: '', name: NO_SUPERVISOR });
// };

// export const getDirectSupervisorId = (
//   dataListPositions: DataValue<API.ListPositionsQuery, {}>,
//   directSupervisors: any[],
//   directSupervisor: string,
// ): string => {
//   const positionItems =
//     dataListPositions && dataListPositions.listPositions && dataListPositions.listPositions.items;
//   const directSupervisorObj = directSupervisors.find(item => item.name === directSupervisor);
//   const directSupervisorObjId = directSupervisorObj ? directSupervisorObj.id : '';
//   const directSupervisorItem =
//     positionItems &&
//     positionItems.find((item: any) => item && item.person && item.person.id === directSupervisorObjId);
//   return directSupervisorItem ? directSupervisorItem.id : '';
// };

export const getPageTitle = (title: string): string => `${title} | Diamond HQ`;

// export const getPersona = (role: string | null | undefined): API.Persona | null => {
//   switch (role) {
//     case 'EMPLOYEE':
//       return API.Persona.EMPLOYEE;
//     case 'CONTRACTOR':
//       return API.Persona.CONTRACTOR;
//     case 'PROJECT_MANAGER':
//       return API.Persona.PROJECT_MANAGER;
//     case 'OPERATIONS_MANAGER':
//       return API.Persona.OPERATIONS_MANAGER;
//     case 'EXECUTIVE':
//       return API.Persona.EXECUTIVE;
//     default:
//       return null;
//   }
// };

// export const isAdmin = (role: string | null): boolean => {
//   const adminRoles: (API.Persona | null)[] = [
//     API.Persona.EXECUTIVE,
//     API.Persona.OPERATIONS_MANAGER,
//     API.Persona.PROJECT_MANAGER,
//   ];
//   if (!adminRoles.includes(getPersona(role))) {
//     return false;
//   }
//   return true;
// };

export enum TimeoffStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  DENIED = 'DENIED',
  CANCELLED = 'CANCELLED',
}

// export const getTimeoffStatus = (
//   statusArray: (API.StatusInput | null)[] | null | undefined,
//   approversArray: (string | null)[] | null | undefined,
// ): TimeoffStatus | undefined => {
//   if (!statusArray || !approversArray) return TimeoffStatus.PENDING;
//   if (statusArray.findIndex(status => status && status.approved === false) > -1)
//     return TimeoffStatus.DENIED;
//   if (statusArray.length < approversArray.length) return TimeoffStatus.PENDING;
//   if (statusArray.length === approversArray.length) return TimeoffStatus.APPROVED;
//   return undefined;
// };

// export const calcCurrentPtoBalance = (
//   items: Array<API.CreateTimeoffInput | null> | null | undefined,
//   trinetPtoBalance: number | null | undefined,
//   firstDateToCompare: Date,
//   lastDateToCompare: Date,
//   allowPending: boolean,
// ): number => {
//   return (
//     items
//       ?.filter(
//         item =>
//           (allowPending ? requestIsOpen(item) : item?.currentStatus === 'APPROVED') &&
//           item?.dateRange &&
//           dateRangeOverlap(
//             new Date(item.dateRange[0].date),
//             new Date(item.dateRange[item.dateRange.length - 1].date),
//             firstDateToCompare,
//             lastDateToCompare,
//           ),
//       )
//       .reduce((balance, item) => {
//         const hrsOfOverlap =
//           item?.dateRange?.reduce((hrs: any, entry: any) => {
//             if (
//               entry.date >= firstDateToCompare.toISOString() &&
//               entry.date <= lastDateToCompare.toISOString()
//             ) {
//               return hrs + 8;
//             }
//             return hrs;
//           }, 0) || 0;
//         return balance - hrsOfOverlap;
//       }, trinetPtoBalance || 0) ||
//     trinetPtoBalance ||
//     0
//   );
// };

export const stableSortDesc = (
  a: { [key: string]: any },
  b: { [key: string]: any },
  orderBy: string,
): number => {
  const aOrderBy = typeof a[orderBy] === 'string' ? a[orderBy].toLowerCase() : a[orderBy];
  const bOrderBy = typeof b[orderBy] === 'string' ? b[orderBy].toLowerCase() : b[orderBy];
  if (!bOrderBy && aOrderBy) return 1;
  if (bOrderBy && !aOrderBy) return -1;
  if (bOrderBy < aOrderBy) {
    return -1;
  }
  if (bOrderBy > aOrderBy) {
    return 1;
  }
  return 0;
};

export const stableSort = (array: any[], cmp: (a: object, b: object) => number): any => {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = cmp(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map(el => el[0]);
};

export const states = [
  'AL',
  'AK',
  'AS',
  'AZ',
  'AR',
  'CA',
  'CO',
  'CT',
  'DE',
  'DC',
  'FM',
  'FL',
  'GA',
  'GU',
  'HI',
  'ID',
  'IL',
  'IN',
  'IA',
  'KS',
  'KY',
  'LA',
  'ME',
  'MH',
  'MD',
  'MA',
  'MI',
  'MN',
  'MS',
  'MO',
  'MT',
  'NE',
  'NV',
  'NH',
  'NJ',
  'NM',
  'NY',
  'NC',
  'ND',
  'MP',
  'OH',
  'OK',
  'OR',
  'PW',
  'PA',
  'PR',
  'RI',
  'SC',
  'SD',
  'TN',
  'TX',
  'UT',
  'VT',
  'VI',
  'VA',
  'WA',
  'WV',
  'WI',
  'WY',
];

// export const transformPersonValue = (
//   value: any,
//   key: string,
// ): API.Persona | string | boolean | null => {
//   switch (key) {
//     case 'role':
//       return getPersona(value);
//     case 'active':
//     case 'billable':
//       return value === 'true' ? true : value === 'false' ? false : null;
//     case 'workEmail':
//       return value.toLowerCase();
//     case 'directSupervisor':
//       return value.id;
//     case 'birthdate':
//       return value ? formatFormDate(value) : null;
//     default:
//       return typeof value === 'string' && value === '' ? null : value;
//   }
// };

export const convertStringToPrimitiveValue = (val: string): boolean | null | undefined => {
  return val === 'true' ? true : val === 'false' ? false : val === 'null' ? null : undefined;
};
