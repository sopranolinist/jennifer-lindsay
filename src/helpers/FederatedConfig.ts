const federatedConfig = {
  google_client_id:
    '935955121454-v27bj0v970740g43e3vghk863jvburum.apps.googleusercontent.com',
  oauth_config: {
    label: 'Sign in with your dws.la Google account'
  }
};
export default federatedConfig;
