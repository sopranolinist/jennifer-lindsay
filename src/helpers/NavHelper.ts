import { matchPath } from 'react-router';
export const drawerWidth = 250;
export const isSelected = (path = '/', currentLocation: string) => {
  const match = matchPath(currentLocation, {
    path,
    exact: true,
    strict: false,
  });
  return !!(match && match.isExact);
};
