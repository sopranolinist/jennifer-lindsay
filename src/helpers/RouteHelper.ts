import { RouteProps } from 'react-router-dom';
import Home from '../screens/Home';
import Contact from '../screens/Contact';
import Portfolio from '../screens/Portfolio';
// import OrgChart from '../screens/OrgChart';
// import Person from '../screens/Person';
// import RecruitingResidualsReport from '../screens/RecruitingResidualsReport';
// import Timeoff from '../screens/Timeoff';
import Unauthorized from '../screens/Unauthorized';

export const activeRoutes = ['home','contact'];
export const routeHome: () => string = () => '/home';
export const routeContact: () => string = () => '/contact';
export const routePortfolio: () => string = () => '/portfolio';
export const routeUnauthorized: () => string = () => '/unauthorized';
export const routeTimeoff: (
  regex: boolean,
  operation?: 'request' | 'list' | 'list-all' | 'list-own',
  rid?: string | null,
) => string = (regex, operation, rid) => {
  const exp = '/timeoff/:operation?/:rid?';

  let path =
    operation || !regex
      ? exp.replace(
          /\/:operation(\??)/,
          typeof operation === 'string' && operation.length ? `/${operation}` : '',
        )
      : exp;
  path =
    rid || !regex
      ? path.replace(/\/:rid(\??)/, typeof rid === 'string' && rid.length ? `/${rid}` : '')
      : path;
  return path;
};

export const routeOrgChart: (
  regex: boolean,
  operation?: 'view' | 'update',
  type?: 'person' | 'position' | 'title',
  uuid?: string | null,
) => string = (regex, operation, type, uuid) => {
  const exp = '/orgchart/:operation?/:type?/:uuid?';

  let path =
    operation || !regex
      ? exp.replace(
          /\/:operation(\??)/,
          typeof operation === 'string' && operation.length ? `/${operation}` : '',
        )
      : exp;
  path =
    type || !regex
      ? path.replace(/\/:type(\??)/, typeof type === 'string' && type.length ? `/${type}` : '')
      : path;
  path =
    uuid || !regex
      ? path.replace(/\/:uuid(\??)/, typeof uuid === 'string' && uuid.length ? `/${uuid}` : '')
      : path;
  return path;
};

export const routePerson: (
  regex: boolean,
  operation?: 'create' | 'list' | 'list-all',
  uuid?: string | null,
) => string = (regex, operation, uuid) => {
  const exp = '/person/:operation?/:uuid?';

  let path =
    operation || !regex
      ? exp.replace(
          /\/:operation(\??)/,
          typeof operation === 'string' && operation.length ? `/${operation}` : '',
        )
      : exp;
  path =
    uuid || !regex
      ? path.replace(/\/:uuid(\??)/, typeof uuid === 'string' && uuid.length ? `/${uuid}` : '')
      : path;
  return path;
};

export const PrivateScreenRoutes: Array<RouteProps> = [
  {
    path: routeHome(),
    component: Home,
    exact: true,
  },
  {
    path: routePortfolio(),
    component: Portfolio,
    exact: true,
  },
  {
    path: routeContact(),
    component: Contact,
    exact: true,
  },
  // {
  //   path: routePerson(true),
  //   component: Person,
  //   exact: true,
  // },
  // {
  //   path: routeTimeoff(true),
  //   component: Timeoff,
  //   exact: true,
  // },
  // {
  //   path: routeOrgChart(true),
  //   component: OrgChart,
  //   exact: true,
  // },
  // {
  //   path: routeRecruitingResidualsReport(),
  //   component: RecruitingResidualsReport,
  //   exact: true,
  // },
];

export const PublicScreenRoutes: Array<RouteProps> = [
  {
    path: routeUnauthorized(),
    component: Unauthorized,
    exact: true,
  },
];
