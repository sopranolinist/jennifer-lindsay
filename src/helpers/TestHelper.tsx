export const testHelper = () => {
  return null;
}
// import 'jest-axe/extend-expect';
// import 'jest-dom/extend-expect';
// import { GenerateId } from 'jss';
// import get from 'lodash/get';
// import React, { ReactChild, ReactElement } from 'react';
// import { MockedProvider, MockedResponse } from 'react-apollo/test-utils';
// import JssProvider from 'react-jss/lib/JssProvider';
// import { MemoryRouter as Router } from 'react-router-dom';
// import * as API from '../API';
// import { AuthProvider, AuthProviderProps, fallbackAuthContextCalls } from '../hooks/AuthContext';
// import { LayoutContextExports, LayoutProvider } from '../hooks/LayoutContext';

// const generateClassName: GenerateId = (rule: any, styleSheet) =>
//   `${get(styleSheet, 'options.classNamePrefix')}-${rule.className}`;

// export const getJssWrappedComponent = (children: ReactChild) => (
//   <JssProvider generateClassName={generateClassName}>{children}</JssProvider>
// );

// export const getRouterWrappedComponent = (children: ReactElement, initialRoute = '/') => (
//   <Router initialEntries={[initialRoute]}>{children}</Router>
// );

// export const getApolloWrappedComponent = (
//   children: ReactElement,
//   apolloMocks?: readonly MockedResponse[] | undefined,
// ) => (
//   <MockedProvider mocks={mocks} addTypename={false}>
//     {children}
//   </MockedProvider>
// );

// export const getAuthContextWrappedComponent = (
//   children: ReactElement,
//   props: AuthProviderProps = { client: undefined },
//   authObjProps?: any,
// ) => {
//   let authProps = {
//     // ...fallbackAuthContextCalls,
//     ...props,
//   };
//   if (authObjProps) {
//     authProps = {
//       ...authProps,
//       defaultAuthObj: {
//         ...fallbackAuthContextCalls.defaultAuthObj,
//         ...authProps.defaultAuthObj,
//         ...authObjProps,
//       },
//     };
//   }
//   return <AuthProvider {...authProps}>{children}</AuthProvider>;
// };

// const LayoutContextDefaultExports: LayoutContextExports = {
//   rightDrawerOpen: false,
//   setRightDrawerOpen: jest.fn(),
// };
// export const getLayoutWrappedComponent = (
//   children: ReactElement,
//   contextValues: Partial<LayoutContextExports> = {},
// ) => (
//   <LayoutProvider value={{ ...LayoutContextDefaultExports, ...contextValues }}>
//     {children}
//   </LayoutProvider>
// );

// type OverridesProp = {
//   authContextProps?: AuthProviderProps;
//   authObjProps?: any;
//   layoutContextValues?: Partial<LayoutContextExports>;
//   initialRoute?: string;
//   apolloMocks?: any; // readonly MockedResponse[] | undefined;
// };
// export const getFullyWrappedComponent = (children: ReactElement, overrides?: OverridesProp) => {
//   let authContextProps: AuthProviderProps | undefined;
//   let authObjProps: any;
//   let layoutContextValues: Partial<LayoutContextExports> | undefined;
//   let initialRoute: string | undefined;
//   let apolloMocks: any; // readonly MockedResponse[] | undefined;
//   if (overrides) {
//     authContextProps = get(overrides, 'authContextProps', {
//       client: undefined,
//     });
//     authObjProps = get(overrides, 'authObjProps');
//     layoutContextValues = get(overrides, 'layoutContextValues', {});
//     initialRoute = get(overrides, 'initialRoute', '');
//   }
//   return getJssWrappedComponent(
//     getApolloWrappedComponent(
//       getRouterWrappedComponent(
//         getAuthContextWrappedComponent(
//           getLayoutWrappedComponent(children, layoutContextValues),
//           authContextProps,
//           authObjProps,
//         ),
//         initialRoute,
//       ),
//       apolloMocks,
//     ),
//   );
// };

// export const mockConsole = () => {
//   jest.spyOn(global.console, 'error').mockImplementation(() => {});
// };

// export const authExamples = {
//   signedOut: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.reject(new Error('not authenticated'));
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: false,
//       defaultIsAuthorized: false,
//     },
//   },
//   signedInAuthorized: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'jen.lindsay@dws.la',
//             identities: [],
//             sub: '87f05f4d-479b-4817-b6a7-63b75eb50c9b',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//     },
//   },
//   signedInUnauthorized: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.unauthorized@dws.la',
//             identities: [],
//             sub: '87f05f4d-479b-4817-b6a7-63b75eb50c9b',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: false,
//     },
//   },
//   signedInAuthorizedEmp: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.employee@dws.la',
//             identities:
//               '[{"userId":"117892860558610669905","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669081099}]',
//             sub: 'e01a0c43-e96a-43f1-887c-8c4e393e134a',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'EMPLOYEE',
//         workEmail: 'dhq.employee@dws.la',
//         personalEmail: 'dhq.employee@gmail.com',
//         role: API.Persona.EMPLOYEE,
//       },
//     },
//   },
//   signedInAuthorizedContr: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.contractor@dws.la',
//             identities:
//               '[{"userId":"100487814204179213154","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669312635}]',
//             sub: '6b736166-4965-4f1d-afbb-80f95be84e64',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'CONTRACTOR',
//         workEmail: 'dhq.contractor@dws.la',
//         personalEmail: 'dhq.contractor@gmail.com',
//         role: API.Persona.CONTRACTOR,
//       },
//     },
//   },
//   signedInAuthorizedExec: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.executive@dws.la',
//             identities:
//               '[{"userId":"107559465961084856432","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669434520}]',
//             sub: '94616d74-f916-48d7-af06-90a3c2e694da',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'EXECUTIVE',
//         workEmail: 'dhq.executive@dws.la',
//         personalEmail: 'dhq.executive@gmail.com',
//         role: API.Persona.EXECUTIVE,
//       },
//     },
//   },
//   signedInAuthorizedProjMgr: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.project_manager@dws.la',
//             identities:
//               '[{"userId":"117019102970691830566","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669554053}]',
//             sub: '9194a067-1d37-406d-a1d2-2505eb4072b4',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'PROJECT_MANAGER',
//         workEmail: 'dhq.project_manager@dws.la',
//         personalEmail: 'dhq.project_manager@gmail.com',
//         role: API.Persona.PROJECT_MANAGER,
//       },
//     },
//   },
//   signedInAuthorizedOpsMgr: {
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.operations_manager@dws.la',
//             identities:
//               '[{"userId":"113701768640542213333","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1564431139511}]',
//             sub: 'f9755b9f-b4ab-403c-964d-f1062f121eed',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'OPERATIONS_MANAGER',
//         workEmail: 'dhq.operations_manager@dws.la',
//         personalEmail: 'dhq.operations_manager@gmail.com',
//         role: API.Persona.OPERATIONS_MANAGER,
//       },
//     },
//   },
//   signedInAuthorizedEmpWithId: {
//     // for tests w/ queries that require an id (non-admin version)
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.employee@dws.la',
//             identities:
//               '[{"userId":"117892860558610669905","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669081099}]',
//             sub: 'e01a0c43-e96a-43f1-887c-8c4e393e134a',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'EMPLOYEE',
//         workEmail: 'dhq.employee@dws.la',
//         personalEmail: 'dhq.employee@gmail.com',
//         id: 'd6d78f20-0d7f-4c1c-9ba8-c7b745be8ac1',
//       },
//     },
//   },
//   signedInAuthorizedExecWithId: {
//     // for tests w/ queries that require an id (admin version)
//     authObjProps: {
//       currentAuthenticatedUser: () => {
//         return Promise.resolve({
//           attributes: {
//             email: 'dhq.executive@dws.la',
//             identities:
//               '[{"userId":"107559465961084856432","providerName":"Google","providerType":"Google","issuer":null,"primary":true,"dateCreated":1565669434520}]',
//             sub: '94616d74-f916-48d7-af06-90a3c2e694da',
//           },
//         });
//       },
//     },
//     authContextProps: {
//       client: undefined,
//       defaultSignedIn: true,
//       defaultIsAuthorized: true,
//       defaultPersonInfo: {
//         firstName: 'DHQ',
//         lastName: 'EXECUTIVE',
//         workEmail: 'dhq.executive@dws.la',
//         personalEmail: 'dhq.executive@gmail.com',
//         id: '899b0c6e-b9c2-446a-bf0c-bcc8b7434d20',
//       },
//     },
//   },
// };
