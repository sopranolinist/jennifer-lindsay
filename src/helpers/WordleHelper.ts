export const MAX_TURNS = 6;
export const BLACK_COLOR = "#222222";
export const YELLOW_COLOR = "#c89403";
export const GREEN_COLOR = "#145314";
export const GRAY_COLOR = "#848383";
