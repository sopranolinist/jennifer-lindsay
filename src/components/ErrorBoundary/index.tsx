import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component, ReactChild } from 'react';
import DocumentTitle from 'react-document-title';
import { getPageTitle } from '../../helpers/DataHelper';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: 'baseline',
      padding: theme.spacing(3),
      paddingTop: theme.spacing(12),
      flexGrow: 1,
      flexDirection: 'column',
      display: 'flex',
      textAlign: 'left'
    },
    errorHeader: {
      color: theme.palette.error['main'],
      marginBottom: theme.spacing(2)
    },
    errorMessage: {
      marginBottom: theme.spacing(2)
    }
  });

interface Props extends WithStyles<typeof styles> {
  children?: ReactChild;
  hasError?: string;
}
interface State {
  error: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: false };
  }

  static getDerivedStateFromError(error: Error | null) {
    return { error: true };
  }

  componentDidCatch(error: Error | null, info: object) {
    // console.log('ErrorBoundary.componentDidCatch', error, info);
  }

  render() {
    const pageTitle = 'Error';
    const { error } = this.state;
    const { children, hasError, classes } = this.props;
    if (!error && !hasError) return children;
    return (
      <div className={classes.root}>
        <DocumentTitle title={getPageTitle(pageTitle)} />
        <Typography component="h1" variant="h6" className={classes.errorHeader}>
          Error
        </Typography>
        <Typography
          component="span"
          variant="body1"
          className={classes.errorMessage}
        >
          The site has encountered an error{hasError && ':'}
        </Typography>
        {hasError && (
          <Typography
            component="span"
            variant="body1"
            className={classes.errorMessage}
          >
            {hasError}
          </Typography>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ErrorBoundary);
