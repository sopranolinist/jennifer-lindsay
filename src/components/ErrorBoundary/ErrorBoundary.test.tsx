import { render } from '@testing-library/react';
import React from 'react';
import { getJssWrappedComponent, mockConsole } from '../../helpers/TestHelper';
import ErrorBoundary from './';

describe('Component - ErrorBoundary', () => {
  beforeEach(() => {
    mockConsole();
  });
  it('Renders ErrorBoundary Fallback if error', () => {
    const ErrorComponent = () => {
      throw new Error('Errored!');
    };
    const component = (
      <ErrorBoundary>
        <ErrorComponent />
      </ErrorBoundary>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it('Renders Component if no error', () => {
    const component = (
      <ErrorBoundary>
        <div />
      </ErrorBoundary>
    );
    const { container } = render(getJssWrappedComponent(component));
    expect(container).toMatchSnapshot();
  });
});
