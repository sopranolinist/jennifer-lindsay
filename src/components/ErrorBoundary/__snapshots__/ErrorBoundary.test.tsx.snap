// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ErrorBoundary Renders Component if no error 1`] = `
<div>
  <div />
</div>
`;

exports[`Component - ErrorBoundary Renders ErrorBoundary Fallback if error 1`] = `
<div>
  <div
    class="ErrorBoundary-root-1"
  >
    <h1
      class="MuiTypography-root ErrorBoundary-errorHeader-2 MuiTypography-h6"
    >
      Error
    </h1>
    <span
      class="MuiTypography-root ErrorBoundary-errorMessage-3 MuiTypography-body1"
    >
      The site has encountered an error
    </span>
  </div>
</div>
`;
